File: proof/induction/examples/1/code.py
def raw_sum(n):
    sum = 0
    for i in range(n+1):
        current_value = (i*(i+1))
        sum = sum + current_value
    return sum

result = raw_sum(21)
print(result) #expect 3542

def rawSquaredSum(n):
    rawTotal = 0
    squaredTotal =0
    for i in range(1, n+1):
        rawTotal = rawTotal+i
    squaredTotal = squaredTotal+ (i*i)
    return rawTotal+squaredTotal

print(rawSquaredSum(255)) # 5592320

def chadFn(n):
    nCubed = n**3
    nSquared = n**2
    numerator = nCubed + 3*nSquared + 2*n
    return numerator/3

print(chadFn(876)) #expect 224841752
# Feel free to implement the test function

File: proof/induction/examples/1/code.cpp
#include <iostream>
#include <cmath>
using std::cout;
using std::endl;

int rawSum(int n)
{
    int answer = 0;
    for (int i = 1; i <= n; i++)
    {
        int currentNum = i * (i + 1);
        answer += currentNum;
    }
    return answer;
}

int rawSquareSum(int n)
{
    int rawTotal = 0;
    int squareTotal = 0;
    for (int i = 1; i <= n; i++)
    {
        rawTotal += i;
        squareTotal += (i * i);
    }
    int total = rawTotal + squareTotal;
    return total;
}

int chadFn(int n)
{
    int nSquared = pow(n, 2);
    int nCubed = pow(n, 3);
    int numerator = nCubed + 3 * nSquared + 2 * n;
    return numerator / 3;
}

int main()
{
    unsigned int n = 255; // select your preferred value
    int ans = chadFn(n);
    int count = rawSum(n);
    int sqSum = rawSquareSum(n);
    cout << ans << "\n" << count << "\n" << sqSum << endl;
    // Should show three lines of the same value 5592320
}